tests/zfs-tests/tests/functional/fault/decompress_fault.ksh

```diff
@@ -31,25 +31,39 @@
 
 log_assert "Testing that injected decompression errors are handled correctly"
 
+if is_freebsd; then
+	COMPRESSION=compressed_arc_enabled
+else
+	COMPRESSION=zfs_compressed_arc_enabled
+fi
+
 function cleanup
 {
-	log_must set_tunable64 zfs_compressed_arc_enabled 1
+	log_must set_tunable64 $COMPRESSION 1
 	log_must zinject -c all
 	default_cleanup_noexit
 }
 
 log_onexit cleanup
 
 default_mirror_setup_noexit $DISK1 $DISK2
-log_must set_tunable64 zfs_compressed_arc_enabled 0
+log_must set_tunable64 $COMPRESSION 0
 log_must zfs create -o compression=on $TESTPOOL/fs
 mntpt=$(get_prop mountpoint $TESTPOOL/fs)
-write_compressible $mntpt 32m 1 0 "testfile"
+if is_freebsd; then
+	# Block size of 0 not supported on FreeBSD. 1024k is default
+	write_compressible $mntpt 32m 1 1024k "testfile"
+else
+	write_compressible $mntpt 32m 1 0 "testfile"
+fi
```

Comment:
A `bs` of 0 isn't really desirable on Linux either, I'd suggest changing this to 1024k and keeping the code common.